composition-js/src/__tests__/matchers/toMatchString.ts

```diff
@@ -34,25 +34,26 @@ expect.extend({
     received = deIndent(received);
     const pass = this.equals(expected, received);
     const message = pass
-      ? () => this.utils.matcherHint('toMatchString', undefined, undefined)
-          + '\n\n'
+      ? () => `${this.utils.matcherHint('toMatchString', undefined, undefined)
+      }\n\n`
           + `Expected: not ${this.printExpected(expected)}`
-      : () => {
-        return (
-          this.utils.matcherHint('toMatchString', undefined, undefined,)
-          + '\n\n'
-          + this.utils.printDiffOrStringify(expected, received, 'Expected', 'Received', true));
-      };
-    return {received, expected, message, name: 'toMatchString', pass};
+      : () => (
+        `${this.utils.matcherHint('toMatchString', undefined, undefined)
+        }\n\n${
+          this.utils.printDiffOrStringify(expected, received, 'Expected', 'Received', true)}`);
+    return {
+      received, expected, message, name: 'toMatchString', pass,
+    };
   },
 
   toMatchStringArray(expected: string[], received: string[]) {
     if (expected.length !== received.length) {
-      const message = () => 
-        this.utils.matcherHint('toMatchStringArray', undefined, undefined,)
-          + `\n\nExpected an array of size ${expected.length} but got one of size ${received.length}\n\n`
-          + this.utils.printDiffOrStringify(expected, received, 'Expected', 'Received', true);
-      return {received, expected, message, name: 'toMatchStringArray', pass: false};
+      const message = () => `${this.utils.matcherHint('toMatchStringArray', undefined, undefined)
```

Comment:
Fwiw, I think the rewrites of strings in that file (this one but others as well) make things quite a bit less unreadable (of course, readability is in parts subjective).

I get it's an automatic rewrite, but I think it's dodgy to rewrite multi-components strings automatically like this.

So if it was me, I'd disable whichever rule deal with this at first, and enable it later in its own commit so it's easier to apply/review manual changes when it gets too ugly. In this file for instance, it's probably worth extracting some things into `const` really. Not strongly insisting, it's probably not the end of the world, but I do think it's a step back.

Reply:
Np, I disabled that rule for now.

Reply:
Looks like changes from this rule are still applied throughout unless I'm imagining the rule incorrectly.

Reply:
Sorry, I think too much time passed and I lost the commit where I disabled it.